includes/rest-api/class-sensei-rest-api-question-helpers-trait.php

```diff
@@ -315,6 +336,42 @@ private function get_question_common_properties( WP_Post $question ): array {
 			'shared'      => ! empty( $question_meta['_quiz_id'] ) && count( $question_meta['_quiz_id'] ) > 1,
 			'categories'  => wp_get_post_terms( $question->ID, 'question-category', [ 'fields' => 'ids' ] ),
 		];
+
+		if ( ! empty( $question_meta['_question_media'][0] ) ) {
+			$question_media = $this->get_question_media( (int) $question_meta['_question_media'][0] );
+
+			if ( ! empty( $question_media ) ) {
+				$common_properties['media'] = $question_media;
+			}
+		}
+
+		return $common_properties;
+	}
+
+	/**
+	 * Helper method to get question media.
+	 *
+	 * @param int $question_media_id The attachment id.
+	 *
+	 * @return array Media info. It includes the type, id, url and title.
+	 */
+	private function get_question_media( int $question_media_id ) : array {
+		$question_media = [];
+		$mimetype       = get_post_mime_type( $question_media_id );
+		$attachment     = get_post( $question_media_id );
+
+		if ( $mimetype || null !== $attachment ) {
+			$mimetype_array = explode( '/', $mimetype );
+
+			if ( ! empty( $mimetype_array[0] ) ) {
+				$question_media['type']  = $mimetype_array[0];
+				$question_media['url']   = esc_url( wp_get_attachment_url( $question_media_id ) );
```

Comment:
I think for images, we should use `wp_get_attachment_image_src` along with the legacy filter `sensei_question_image_size`. Otherwise this is migrating the images to using their full size original.

https://github.com/Automattic/sensei/blob/612199ff2bc06039ed64fa3f85130d40b6b75d33/includes/class-sensei-question.php#L629-L639

Reply:
Yeah I thought about this too but I thought that we should get rid of the filter as all image settings would be configured by the block. The user needs to save the lesson for the image to get migrated so it isn't possible to save the image with a wrong size without the user noticing it.

On the other hand this filter will be used only during migration so it isn't an issue if we keep it.

Reply:
Yeah, I think at some point we should deprecate this filter, but for now I think a goal should be when people save a quiz made in the metabox editor using the block editor, it should look the same on the frontend. These changes look good!